modules/collabora_online_group/tests/src/Functional/ViewTest.php

```diff
@@ -0,0 +1,143 @@
+
+namespace Drupal\Tests\collabora_online_group\Functional;
+
+use Drupal\Core\Url;
+use Drupal\group\PermissionScopeInterface;
+use Drupal\media\Entity\Media;
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\collabora_online\Traits\MediaCreationTrait;
+use Drupal\Tests\collabora_online_group\Traits\GroupRelationTrait;
+use Drupal\Tests\group\Traits\GroupTestTrait;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+use Drupal\user\RoleInterface;
+
+/**
+ * Test the provided view by the module.
+ */
+class ViewTest extends BrowserTestBase {
+
+  use MediaCreationTrait;
+  use GroupRelationTrait;
+  use GroupTestTrait;
+  use MediaTypeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'collabora_online_group',
+    'views',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Test the links for Collabora operations in the view.
+   */
+  public function testViewLinks(): void {
+    // Add configuration needed for testing.
+    $group_type = $this->createGroupType(['id' => 'group_type_1']);
+    $media_type = $this->createMediaType('file', ['id' => 'document']);
+    $this->createGroupRole([
+      'group_type' => 'group_type_1',
+      'scope' => PermissionScopeInterface::INSIDER_ID,
+      'global_role' => RoleInterface::AUTHENTICATED_ID,
+      'permissions' => [
+        'view group',
+        'access group_media overview',
+        'view group_media:document entity',
+        'edit any group_media:document in collabora',
+        'preview group_media:document in collabora',
+        ],
+    ]);
+    $this->createPluginRelation($group_type, 'group_media:document', [
+      'group_cardinality' => 0,
+      'entity_cardinality' => 1,
+      'use_creation_wizard' => FALSE,
+    ]);
+
+    // Create content.
+    $group = $this->createGroup(['type' => 'group_type_1']);
+    for ($i = 1;$i < 4;$i++) {
+      $media = $this->createMediaEntity('document', [
+        'id' => 'media_' . $i,
+        'name' => 'Media ' . $i,
+      ]);
+      $group->addRelationship($media, 'group_media:document');
+    }
+    $user = $this->createUser([
+      'view the administration theme',
+      'access administration pages',
+      'access group overview',
+    ]);
+    $group->addMember($user);
+
+    // Go to the page and check the links added to the view.
+    $this->drupalLogin($user);
+    $this->drupalGet("group/{$group->id()}/media");
+    $assert_session = $this->assertSession();
+
+    // Check table header.
+    $table = $assert_session->elementExists('css', 'table');
+    $table_header = $assert_session->elementExists('css', 'thead', $table);
+    $rows = $table_header->findAll('css', 'tr');
+    $cols = $rows[0]->findAll('css', 'th');
+    $this->assertEquals('Media name', $cols[0]->getText());
+    $this->assertEquals('Bundle', $cols[1]->getText());
+    $this->assertEquals('Status', $cols[2]->getText());
+    $this->assertEquals('Publisher', $cols[3]->getText());
```

Comment:
I like combined assertions like
```php
$header = array_map(fn ($col) => $col->getText(), $cols);
$this->assertSame(['Media name', 'Bundle', 'Status', 'Publisher', 'Operations'], $header);
```
This contradicts a bit my other suggestion below where we need the special treatment for the last column header. So maybe in this case we don't do it.